public/components/data_connections/components/query_permissions_flex_item.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiComboBox,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiRadioGroup,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import React from 'react';
+import { PermissionsFlexItem } from '../../../../common/types/data_connections';
+import {
+  OPENSEARCH_DOCUMENTATION_URL,
+  QUERY_RESTRICT,
+} from '../../../../common/constants/data_connections';
+
+export const QueryPermissionsFlexItem = (props: PermissionsFlexItem) => {
+  const { roles, setSelectedRoles, selectedRoles, selectedRadio, radios, setSelectedRadio } = props;
+  return (
+    <EuiFlexItem>
+      <EuiFlexGroup direction="row">
+        <EuiFlexItem>
+          <EuiText className="overview-title">Query Permissions</EuiText>
+          <EuiText size="s" className="overview-content">
+            Control which OpenSearch roles have query permissions on this data source.{' '}
+            <EuiLink external={true} href={OPENSEARCH_DOCUMENTATION_URL} target="_blank">
+              Learn more
+            </EuiLink>
+          </EuiText>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiRadioGroup
+            options={radios}
+            idSelected={selectedRadio}
+            onChange={(id) => setSelectedRadio(id)}
+            name="query-radio-group"
+            legend={{
+              children: <span>Access level</span>,
+            }}
+          />
+          {selectedRadio === QUERY_RESTRICT ? (
```

Comment:
Consider moving this condition to a component-constant.  Name it something that tells the story about what it is / why it's there.

We are left to imagine the meaning of  `selectedRadio === QUERY_RESTRICT`... often the wrong meaning.

Reply:
Not quite sure I am following what you are suggesting here. The issue is that right now, the radio groups and modifying the radio groups need to be passed down from the parent to this specific component, but this component also needs to know some specifics about the radio groups in order to render correctly. This situation sounds like maybe the parent should not be doing that, and should just allow this component to render everything and control as needed. However this means that the parent needs to pass down the whole data connection details so that this child component can make the appropriate API call to update it. Does this tradeoff make sense?